### PR TITLE
First pass at simple shared memory IPC

### PR DESCRIPTION
**Pull request by benkroop**
This should unblock us, but there's a lot more work I need to do.

### CODE REVIEW

File: infrastructure/comms/shared_struct/shared_struct_publisher.inl [hunk 1]
```diff
@@ -0,0 +1,43 @@
+
+namespace jet {
+
+SharedStructPublisher::SharedStructPublisher(const std::string& object_name) : Publisher(object_name) {
+  int o_flags = O_RDWR | O_CREAT;
+  off_t length = 921650 + 1024;
```
Comment (jpanikulam) [inline infrastructure/comms/shared_struct/shared_struct_publisher.inl]: Consider making these named constants (and making them `const`)

File: infrastructure/comms/shared_struct/shared_struct_publisher.inl [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  int shmem_file_descriptor_ = shm_open(object_name.c_str(), o_flags, 0644 );
+  if (shmem_file_descriptor_ < 0) {
+    std::cerr << "Error opening shared memory region: " <<  errno << std::endl;
+    throw std::runtime_error("Could not open shared memory region.");
```
Comment (jpanikulam) [inline infrastructure/comms/shared_struct/shared_struct_publisher.inl]: Consider only throwing the error (and not printing)

File: infrastructure/comms/shared_struct/shared_struct_subscriber.inl [hunk 3]
```diff
@@ -0,0 +1,55 @@
+
+namespace jet {
+
+SharedStructSubscriber::SharedStructSubscriber(const std::string& object_name) : Subscriber(object_name) {
+  int o_flags = O_RDWR;
+  int shmem_file_descriptor_ = shm_open(object_name.c_str(), o_flags, 0644);
```
Comment (jpanikulam) [inline infrastructure/comms/shared_struct/shared_struct_subscriber.inl]: Same about named constants

Comment (benkroop) [reply infrastructure/comms/shared_struct/shared_struct_subscriber.inl]: :+1:

File: infrastructure/comms/shared_struct/shared_struct_publisher.inl (hunk 2 above)
Comment (benkroop) [reply infrastructure/comms/shared_struct/shared_struct_publisher.inl]: :+1:

File: infrastructure/comms/shared_struct/shared_struct_publisher.inl (hunk 1 above)
Comment (benkroop) [reply infrastructure/comms/shared_struct/shared_struct_publisher.inl]: :+1:

**Pull request merged**